src/sprite/sprite_data.cpp

```diff
@@ -34,10 +35,64 @@
 #include "video/surface.hpp"
 #include "video/texture_manager.hpp"
 
+LinkedSpritesContainer::LinkedSpritesContainer() :
+  linked_light_sprite(),
+  linked_sprites()
+{
+}
+
+void
+LinkedSpritesContainer::parse_linked_sprites(const ReaderMapping& mapping)
+{
+  auto iter_sprites = mapping.get_iter();
+  while (iter_sprites.next())
+  {
+    const auto& sx = iter_sprites.as_mapping().get_sexp();
+    const auto& arr = sx.as_array();
+
+    std::string filepath = FileSystem::join(mapping.get_doc().get_directory(), arr[1].as_string());
+    if (!PHYSFS_exists(filepath.c_str())) // If file path is not relative to current directory, make it relative to root
+      filepath = arr[1].as_string();
+
+    const std::string key = arr[0].as_string();
+    if (key == "light") // The key "light" is reserved for light sprites
+    {
+      linked_light_sprite = LinkedLightSprite(filepath);
+
+      if (arr.size() >= 3) // Default action has been specified
+      {
+        linked_light_sprite->action = arr[2].as_string();
+
+        if (arr.size() >= 6) // Color has been specified
+        {
+          linked_light_sprite->color = Color(arr[3].as_float(), arr[4].as_float(),
+                                             arr[5].as_float());
+        }
+      }
```

Comment:
You don't need to nest these
```suggestion
      if (arr.size() >= 3) // Default action has been specified
      {
        linked_light_sprite->action = arr[2].as_string();
      }
      
      if (arr.size() >= 6) // Color has been specified
      {
        linked_light_sprite->color = Color(arr[3].as_float(), arr[4].as_float(),
                                           arr[5].as_float());
      }
```

Reply:
I do, if it's not higher than or equal to 3, it can't be higher than or equal to 6.